lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -41,6 +41,9 @@ def create_vm(attrs)
             attrs[:comment] ||= ""
             attrs[:quota] = attrs[:quota].present? ? client.system_service.data_centers_service.data_center_service(datacenter).quotas_service.quota_service(attrs[:quota]).get : nil
 
+            cpu_topology = OvirtSDK4::CpuTopology.new(:cores => attrs[:cores], :sockets => "1") if attrs[:cores].present?
+            attrs[:cpu] = OvirtSDK4::Cpu.new(:topology => cpu_topology)
```

Comment:
I'd be slightly inclined towards not assigning anything to `attrs[:cpu]` in case `attrs[:cores]` is `#blank?` and letting `OvirtSDK4` fill in the defaults, but from what I've seen in OSDK4 it shouldn't really matter.

+0 for having this done conditionally